tests/bgp/test_4-byte_asn_community.py

```diff
@@ -0,0 +1,243 @@
+    -c "address-family ipv4 unicast" \
+    -c "network {}" \
+    -c "neighbor {} soft-reconfiguration inbound" \
+    -c "neighbor {} route-map FROM_BGP_PEER_V4 in" \
+    -c "neighbor {} route-map TO_BGP_PEER_V4 out" \
+    -c "neighbor {} activate" \
+    -c "maximum-paths 64" \
+    -c "exit-address-family" \
+    -c "address-family ipv6 unicast" \
+    -c "network {}" \
+    -c "neighbor {} soft-reconfiguration inbound" \
+    -c "neighbor {} route-map FROM_BGP_PEER_V6 in" \
+    -c "neighbor {} route-map TO_BGP_PEER_V6 out" \
+    -c "neighbor {} activate" \
+    -c "maximum-paths 64" \
+    -c "exit-address-family" \
+    '.format(setup['neigh_cli_options'], setup['neigh_asn'], neighbor_4byte_asn, setup['neigh_bgp_id'],
+             setup['peer_group_v4'], setup['peer_group_v6'], setup['dut_ip_v4'], dut_4byte_asn, setup['dut_ip_v4'],
+             setup['peer_group_v4'], setup['dut_ip_v4'], 'DUT', setup['dut_ip_v4'], setup['dut_ip_v4'],
+             setup['dut_ip_v6'], dut_4byte_asn, setup['dut_ip_v6'], setup['peer_group_v6'], setup['dut_ip_v6'], 'DUT',
+             setup['dut_ip_v6'], setup['dut_ip_v6'], setup['neigh_ipv4_network'], setup['peer_group_v4'],
+             setup['peer_group_v4'], setup['peer_group_v4'], setup['dut_ip_v4'], setup['neigh_ipv6_network'],
+             setup['peer_group_v6'], setup['peer_group_v6'], setup['peer_group_v6'], setup['dut_ip_v6'])
+
+    logger.debug(setup['neighhost'].shell(cmd, module_ignore_errors=True))
+
+    logger.info("DUT BGP Config: {}".format(setup['duthost'].shell("show run bgp")['stdout']))
+    logger.info("Neighbor BGP Config: {}".format(setup['neighhost'].shell("show run bgp")['stdout']))
+
+    time.sleep(bgp_sleep)
+
+    output = setup['duthost'].shell("show ip bgp summary | grep {}".format(setup['neigh_ip_v4']))['stdout']
+    assert str(neighbor_4byte_asn) in output
+    output = setup['duthost'].shell("show ipv6 bgp summary | grep {}".format(setup['neigh_ip_v6'].lower()))['stdout']
+    assert str(neighbor_4byte_asn) in output
+    output = setup['duthost'].shell("show ip bgp neighbors {} routes".format(setup['neigh_ip_v4']))['stdout']
+    assert str(neighbor_4byte_asn) in output
+    output = setup['duthost'].shell("show ipv6 bgp neighbors {} routes".format(setup['neigh_ip_v6'].lower()))['stdout']
+    assert str(neighbor_4byte_asn) in output
+
+    output = setup['neighhost'].shell("show ip bgp summary | grep {}".format(setup['dut_ip_v4']))['stdout']
+    assert str(dut_4byte_asn) in output
+    output = setup['neighhost'].shell("show ipv6 bgp summary | grep {}".format(setup['dut_ip_v6'].lower()))['stdout']
+    assert str(dut_4byte_asn) in output
+    output = setup['neighhost'].shell("show ip bgp neighbors {} routes".format(setup['dut_ip_v4']))['stdout']
+    assert str(dut_4byte_asn) in output
+    output = setup['neighhost'].shell("show ipv6 bgp neighbors {} routes".format(setup['dut_ip_v6'].lower()))['stdout']
+    assert str(dut_4byte_asn) in output
```

Comment:
The asn can match anywhere in the output resulting in the test to pass - though less likely. The right way is to parse the route and match asn where the as-path is present after the prefix.

Reply:
@hasan-brcm I updated to match in a specific place of the output now.